### PR TITLE
Visitor canvas

### PR DESCRIPTION
**Pull request by JakZaj**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/canvas/Canvas.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+public abstract class Canvas {
+	protected int width;
+	protected int height;
+
+	public Canvas(int width, int height) {
+		this.width = width;
+		this.height = height;
+	}
+
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/visitor/canvas/Canvas.java]: This is a RectangularCanvas. Please extract also a sole interface to enable support of wider range of canvases.

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/canvas/A4Canvas.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package edu.kis.powp.jobs2d.command.visitor.canvas;
+
+public class A4Canvas extends Canvas {
+	public A4Canvas() {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/visitor/canvas/A4Canvas.java:4]: A4Canvas looks like a specific version of CustomCanvas (superclass?)

File: src/main/java/edu/kis/powp/jobs2d/command/visitor/canvas/IsInBoundsVisitor.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+		}
+	}
+
+	@Override
+	public void visit(ICompoundCommand compoundCommand) {
+		Iterator<DriverCommand> iterator = compoundCommand.iterator();
+		while (iterator.hasNext()) {
+			DriverCommand driverCommand = iterator.next();
+			driverCommand.accept(this);
+		}
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/visitor/canvas/IsInBoundsVisitor.java]: This approach is not reusable (visitor can be called once). If this method kept local isInBounds varialble the other visit methods could be simpler:
isInBounds = canvas.isInBounds(setPositionCommand.getPosX(),setPositionCommand.getPosY());

**Review by kowallus — CHANGES_REQUESTED** [review]
small refactor suggestions

**Review by 209424 — APPROVED** [review]

**Review by kowallus — APPROVED** [review]
Nice :)

**Pull request merged**